src/lib/select/select.md

```diff
@@ -1,111 +1,166 @@
+include error messages, hint text, prefix & suffix, and theming. For additional information about
+these features, see the
+[form field documentation](https://material.angular.io/components/form-field/overview).
+
+<!-- example(select-hint-error) -->
+
+### Setting a static placeholder
+
+The placeholder is a text label displayed in the select trigger area when no value is selected. When
+a value is selected, the placeholder will float above the select trigger area. The placeholder can
+be specified either via a `placeholder` attribute on the `<mat-select>` or a `<mat-placeholder>`
+element in the same form field as the `<mat-select>`. The `<mat-form-field>` also has additional
+options for changing the behavior of the placeholder. For more information see the
+[form field placeholder documentation](https://material.angular.io/components/form-field/overview#floating-placeholder).
+
+### Disabling the select or individual options
+
+It is possible to disable the entire select or individual options in the select by using the
+disabled property on the `<mat-select>` and the `<mat-option>` components respectively.
+
+<!-- example(select-disabled) -->
 
 ### Resetting the select value
 
-If you want one of your options to reset the select's value, you can omit specifying its value:
+If you want one of your options to reset the select's value, you can omit specifying its value.
 
-*my-comp.html*
-```html
-<mat-form-field>
-  <mat-select placeholder="State">
-    <mat-option>None</mat-option>
-    <mat-option *ngFor="let state of states" [value]="state.code">{{ state.name }}</mat-option>
-  </mat-select>
-</mat-form-field>
-```
+<!-- example(select-reset) -->
 
-### Setting a static placeholder
+### Creating groups of options
 
-It's possible to turn off the placeholder's floating animation using the `floatPlaceholder` property. It accepts one of three string options:
-- `'auto'`: This is the default floating placeholder animation. It will float up when a selection is made.
-- `'never'`: This makes the placeholder static. Rather than floating, it will disappear once a selection is made.
-- `'always'`: This makes the placeholder permanently float above the input. It will not animate up or down.
+The `<mat-optgroup>` element can be used to group common options under a subheading. The name of the
+group can be set using the `label` property of `<mat-optgroup>`. Like individual `<mat-option>`
+elements, an entire `<mat-optgroup>` can be disabled or enabled by setting the `disabled` property
+on the group.
 
-```html
-<mat-form-field floatPlaceholder="never">
-  <mat-select placeholder="State" [(ngModel)]="myState">
-  <mat-option *ngFor="let state of states" [value]="state.code">{{ state.name }}</mat-option>
-  </mat-select>
-</mat-form-field>
-```
+<!-- example(select-optgroup) -->
+
+### Selecting multiple options
+
+By default, the `<mat-select>` only supports selecting a single option at a time. However, you can 
```

Comment:
"By default, the `<mat-select>` only supports selecting a single option at a time. "
->
"`<mat-select>` defaults to single-selection mode, but can be configured..."